hazelcast-sql/src/main/java/com/hazelcast/sql/impl/calcite/parse/UnsupportedOperationVisitor.java

```diff
@@ -204,6 +208,10 @@ public Void visit(SqlDataTypeSpec type) {
             case REAL:
             case DOUBLE:
             case VARCHAR:
+            case DATE:
+            case TIME:
+            case TIMESTAMP:
+            case TIME_WITH_LOCAL_TIME_ZONE:
```

Comment:
What's the status of TIME_WITH_LOCAL_TIME_ZONE? There is no mapping from/to query data for it, but it's listed as allowed.

Reply:
This is a bug, neither `TIME_WITH_LOCAL_TIME_ZONE`, nor `TIMESTAMP_WITH_LOCAL_TIME_ZONE` should be there. We should list there only built-in Calcite types that we use. Our custom `TIMESTAMP_WITH_TIME_ZONE` is not in the list, because we rely on custom identifier remapping here (similarly to `ANY` -> `OBJECT`)

Fixed in https://github.com/hazelcast/hazelcast/pull/17395/commits/5ebe4af6c3654ab9a3f52666d25517929e7e94e5